### PR TITLE
LG-2521 Unified `InternalInputOption` component

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Creates an internal InputOption component, to share styles and functionality between various components like Combobox, Select, SearchInput and Menu

🎟 _Jira ticket:_ [LG-2521](https://jira.mongodb.org/browse/LG-2521)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/internal-input-option/src/InputOption.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import { InputOption } from '.';
+
+describe('packages/internal-input-option', () => {
```
Comment (brookescarlett) [inline packages/internal-input-option/src/InputOption.spec.tsx:5]: feels like we should be testing a little more here...

`className` prop appears in the classLIst, `onClick` gets called, `disabled` works, etc.

Comment (brookescarlett) [reply packages/internal-input-option/src/InputOption.spec.tsx:5]: children render

**Review by brookescarlett — COMMENTED** [review]
Inclined to let this in sooner rather than later as it's largely internal and I think your next component work will be a good test of whether or not this does what we need it to. We're going to have to create tickets to port the other components to use this as well

**Comment by brookescarlett** [comment]
Need to update to follow new directory structure

**Comment by TheSonOfThomp** [comment]
Closing in favor of #1528

**Pull request closed without merging**